datadog-sidecar/src/service/telemetry.rs

```diff
@@ -111,6 +113,7 @@ impl TelemetryCachedClient {
             &self.config_sent,
             &self.buffered_integrations,
             &self.buffered_composer_paths,
+            &self.last_endpoints_push,
```

Comment:
I don't think you can with this guarantee that you have only one PHP worker submitting endpoints. Both workers can read an old value for `last_endpoints_push` and submit the data.

I think you have a few options. Without analyzing their easiness/viability:
* Query last_endpoints_push not through this read-only shared memory, but instead through a Sidecar call than atomically swap last_endpoints_push
* Keep track of which session wrote the first endpoint, and reject endpoint submissions from other sessions
* Submit the endpoints instead of one by one, all at once, replacing whatever was stored.
* Keep the endpoints in a set, and accept them being submitted more than once

Reply:
I change the store of endpoints to a hashset